migrator/version/version.go

```diff
@@ -0,0 +1,93 @@
+package version
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/migrator/log"
+	migSchema "github.com/stackrox/rox/migrator/migrations/postgreshelper/schema"
+	migGorm "github.com/stackrox/rox/migrator/postgres/gorm"
+	"github.com/stackrox/rox/pkg/migrations"
+	"github.com/stackrox/rox/pkg/postgres/pgutils"
+	pkgSchema "github.com/stackrox/rox/pkg/postgres/schema"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/timestamp"
+	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stackrox/rox/pkg/version"
+	"gorm.io/gorm"
+)
+
+// ReadVersionPostgres - reads the version from the postgres database.
+func ReadVersionPostgres(dbName string) (*migrations.MigrationVersion, error) {
+	gc := migGorm.GetConfig()
+
+	ver := migrations.MigrationVersion{MainVersion: "0", SeqNum: 0}
+	db, err := gc.ConnectWithRetries(dbName)
+	if err != nil {
+		return &ver, nil
+	}
+	defer migGorm.Close(db)
+	return ReadVersionGormDB(db)
+}
+
+// ReadVersionGormDB - reads the version from the postgres database with a gorm instance.
+func ReadVersionGormDB(db *gorm.DB) (*migrations.MigrationVersion, error) {
+	ctx := sac.WithAllAccess(context.Background())
+	pkgSchema.ApplySchemaForTable(ctx, db, pkgSchema.VersionsSchema.Table)
+	var modelVersion pkgSchema.Versions
+	ver := migrations.MigrationVersion{MainVersion: "0", SeqNum: 0}
+	result := db.Table(pkgSchema.VersionsSchema.Table).First(&modelVersion)
+	if result.Error != nil {
+		return &ver, nil
+	}
+
+	protoVersion, err := migSchema.ConvertVersionToProto(&modelVersion)
+	if err != nil {
+		return &ver, nil
+	}
+
+	log.WriteToStderrf("Migration version from DB = %s.", protoVersion)
+
+	ver.MainVersion = protoVersion.GetVersion()
+	ver.SeqNum = int(protoVersion.GetSeqNum())
+	ver.LastPersisted = timestamp.FromProtobuf(protoVersion.GetLastPersisted()).GoTime()
+	return &ver, nil
+}
+
+// SetVersionPostgres - sets the version in the named postgres database
+func SetVersionPostgres(dbName string, updatedVersion *storage.Version) {
+	db, err := migGorm.GetConfig().ConnectWithRetries(dbName)
+	if err != nil {
+		utils.Must(errors.Wrapf(err, "failed to connect to database %s", dbName))
+	}
+	defer migGorm.Close(db)
+	SetVersionGormDB(db, updatedVersion)
+}
+
+// SetVersionGormDB - sets the version in the postgres database specified with the Gorm instance
+func SetVersionGormDB(db *gorm.DB, updatedVersion *storage.Version) {
+	ctx := sac.WithAllAccess(context.Background())
+	pkgSchema.ApplySchemaForTable(ctx, db, pkgSchema.VersionsSchema.Table)
```

Comment:
What is the cost of this operation when the schema has already been applied?  Reason I ask is that this will get called for each migration as we loop through and process them.

Reply:
It should not be much but it is always good to avoid that. Thanks.